### PR TITLE
Handle unready Caches

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
Currently when the master-controller-manager starts up and acquired its leader lease, the usersshkeysyncer and projectlabelsyncer controllers will attempt to reconcile as soon as possible. This causes problems because the caches have not yet been started, so the pod is logging a bunch of serious sounding errors with stacktraces. This is especially misleading when debugging something, because you fetch the pod's logs and are greeted with many errors.

This PR extends the two controllers a bit to

1. Handle the worker-name labels as early as possible. This is not strictly required to fix the errors, but makes the controller play nicer when other Kubermatic installations are in the same cluster (as we now have due to the Operator and me testing it on dev).
2. Return the `ErrCacheNotStarted` so that we can handle it gracefully and simply retry.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/util/util.go [hunk 1]
```diff
@@ -144,3 +151,14 @@ func ConcurrencyLimitReached(ctx context.Context, client ctrlruntimeclient.Clien
...
 	return clustersUpdatingInProgressCount >= limit, nil
 }
+
+// IsCacheNotStarted returns true if the given error is not nil and an instance of
+// cache.ErrCacheNotStarted.
+func IsCacheNotStarted(err error) bool {
+	if err == nil {
+		return false
+	}
+
```
Comment (xrstf) [inline api/pkg/controller/util/util.go:161]: I'm not 100000% sure that we need this nil check. But it feels safe to have it.

Comment (kron4eg) [reply api/pkg/controller/util/util.go:161]: true, it's absolutely excessive: https://play.golang.org/p/vf1uN8EjVHB

**Review by kron4eg — APPROVED** [review]
/lgtm
/approve

/hold

@xrstf there is microscopic nil, if you don't feel like removing it, then please simply unhold the PR

**Pull request merged**